Android/ConstraintLayout/build.cake

```diff
@@ -1,35 +1,65 @@
-#load "../../common.cake"
+var TARGET = Argument ("target", Argument ("t", "ci"));
 
-var TARGET = Argument ("target", Argument ("t", "Default"));
-
-var NUGET_VERSION = "2.0.0-alpha2";
-var AAR_VERSION = "2.0.0-alpha2";
+var NUGET_VERSION = "2.0.0-beta2";
+var AAR_VERSION = "2.0.0-beta2";
 
 var CONSTRAINT_LAYOUT_URL = string.Format ("https://dl.google.com/dl/android/maven2/com/android/support/constraint/constraint-layout/{0}/constraint-layout-{0}.aar", AAR_VERSION);
 var CONSTRAINT_LAYOUT_SOLVER_URL = string.Format ("https://dl.google.com/dl/android/maven2/com/android/support/constraint/constraint-layout-solver/{0}/constraint-layout-solver-{0}.jar", AAR_VERSION);
 
+Task("libs")
+	.IsDependentOn("externals")
+	.Does (() =>
+{
+	MSBuild("./source/ConstraintLayout.sln", c => {
+		c.Configuration = "Release";
+		c.Targets.Clear();
+		c.Targets.Add("Restore");
+		c.Targets.Add("Rebuild");
+		c.Properties.Add("DesignTimeBuild", new [] { "false" });
+		c.BinaryLogger = new MSBuildBinaryLogSettings {
+			Enabled = true,
+			FileName = "./output/libs.binlog"
+		};
+	});
+});
+
+Task("samples")
+	.IsDependentOn("nuget")
+	.Does (() =>
+{
+	MSBuild("./samples/ConstraintLayoutSample.sln", c => {
+		c.Configuration = "Release";
+		c.Targets.Clear();
+		c.Targets.Add("Restore");
+		c.Targets.Add("Build");
+		c.BinaryLogger = new MSBuildBinaryLogSettings {
+			Enabled = true,
+			FileName = "./output/samples.binlog"
+		};
+	});
+});
 
-var buildSpec = new BuildSpec {
-	Libs = new [] {
-		new DefaultSolutionBuilder {
-			SolutionPath = "./source/ConstraintLayout.sln",
-			BuildsOn = BuildPlatforms.Windows | BuildPlatforms.Mac,
-			OutputFiles = new [] {
-				new OutputFileCopy { FromFile = "./source/ConstraintLayout/bin/Release/Xamarin.Android.Support.Constraint.Layout.dll", ToDirectory = "./output/" },
-				new OutputFileCopy { FromFile = "./source/ConstraintLayoutSolver/bin/Release/Xamarin.Android.Support.Constraint.Layout.Solver.dll", ToDirectory = "./output/" },
-			}
-		}
-	},
 
-	Samples = new [] {
-		new DefaultSolutionBuilder { SolutionPath = "./samples/ConstraintLayoutSample.sln", BuildsOn = BuildPlatforms.Windows | BuildPlatforms.Mac },
-	},
+Task("nuget")
+	.IsDependentOn("libs")
+	.Does (() =>
+{
+	MSBuild ("./source/ConstraintLayout.sln", c => {
+		c.Configuration = "Release";
+		c.Targets.Clear();
+		c.Targets.Add("Pack");
```

Comment:
One thing that you can do to speed up the pack and not trigger a build (because you just built it already) is to add the `NoBuild` property: `c.Properties.Add("NoBuild", new [] { "true" });`

Reply:
It really doesn't add much time to it, I think we should KISS and leave as is, but good to know!